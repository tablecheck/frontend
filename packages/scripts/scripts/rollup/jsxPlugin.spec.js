const rollup = require('rollup');
const { jsxPlugin } = require('./jsxPlugin');

function testCode(name, filePath) {
  // eslint-disable-next-line jest/valid-title
  test(name, async () => {
    const resolvedFilePath = require.resolve(filePath);
    const bundle = await rollup.rollup({
      input: resolvedFilePath,
      plugins: [
        {
          name: 'exclude-assets',
          resolveId(id) {
            if (id === resolvedFilePath) return null;
            // pretend all imports are external
            return { id, external: true };
          }
        },
        jsxPlugin()
      ]
    });
    const generated = await bundle.generate({ format: 'esm' });
    expect(generated.output[0].code).toMatchSnapshot();
  });
}

describe('scripts/jsxPlugin', () => {
  testCode('should handle basic JSX elements', './fixtures/basic.jsx');

  testCode(
    'should handle varying import patterns',
    './fixtures/importedComponents.jsx'
  );

  testCode('should handle name clashes', './fixtures/variableNameClashes.jsx');

  testCode('should handle import alias spread', './fixtures/aliasSpread.jsx');

  testCode('should handle emotion css', './fixtures/emotionCss.jsx');
});
