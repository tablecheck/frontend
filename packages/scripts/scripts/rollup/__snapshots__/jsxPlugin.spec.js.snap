// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scripts/jsxPlugin should handle basic JSX elements 1`] = `
"import { jsxs, jsx } from '@emotion/react/jsx-runtime';

function Component() {
  return jsxs(\\"div\\", {
    children: [jsx(\\"link\\", {
      type: \\"self closing\\"
    }), jsx(\\"p\\", {
      children: jsx(\\"a\\", {
        href: \\"123\\",
        children: \\"Link Content\\"
      })
    })]
  });
}

export { Component };
"
`;

exports[`scripts/jsxPlugin should handle emotion css 1`] = `
"import { css } from '@emotion/react';
import { getThemeValue } from '@tablekit/utils';
import { tooltipThemeNamespace, tooltipClassicTheme } from './themes';

const TooltipPrimitive = styled.div\`
  pointer-events: none;
  position: fixed;
\`;
const Tooltip = styled(TooltipPrimitive)\`
  background-color: \${getThemeValue(\`\${tooltipThemeNamespace}.backgroundColor\`, tooltipClassicTheme.backgroundColor)};
  color: \${getThemeValue(\`\${tooltipThemeNamespace}.textColor\`, tooltipClassicTheme.textColor)};
  top: 0;
  left: 0;
  line-height: 16px;
  max-width: 240px;

  \${({
  isSelected,
  theme
}) => {
  if (isSelected) {
    return css(\\"color:white;background-color:\\", theme.colors.primary, \\";&:hover{background-color:\\", theme.colors.primary2, \\";}\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\";label:Tooltip;\\"));
  }

  return '';
}}
\`;

export { Tooltip, TooltipPrimitive };
"
`;

exports[`scripts/jsxPlugin should handle import alias spread 1`] = `
"import { components } from 'my-select';
import { components as components$1 } from 'react-select';
import { jsx } from '@emotion/react/jsx-runtime';

function createSelect(props) {
  const {
    MyControl
  } = components;
  const {
    Control
  } = components$1;
  return jsx(Control, { ...props,
    children: jsx(MyControl, {
      children: jsx(\\"div\\", {
        children: \\"Test\\"
      })
    })
  });
}

export { createSelect };
"
`;

exports[`scripts/jsxPlugin should handle name clashes 1`] = `
"import { InputLabelWrapper } from './styled/label';

const Label = ({
  label
}) => InputLabelWrapper(label);

export { Label };
"
`;

exports[`scripts/jsxPlugin should handle varying import patterns 1`] = `
"import { Component } from './component';
import { components } from './components';
import DefaultComponent from './defaults';
import { jsxs, jsx } from '@emotion/react/jsx-runtime';

function TestComponent() {
  return jsxs(\\"div\\", {
    children: [jsx(Component, {
      prop1: 12
    }), jsx(components.SomeProp, {
      children: \\"Children\\"
    }), jsx(DefaultComponent, {
      children: jsx(\\"div\\", {
        children: \\"Excess\\"
      })
    })]
  });
}

export { TestComponent as default };
"
`;
